src/Nox.Generator/Application/DomainEventHandlers/DomainEventHandlerGenerator.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Microsoft.CodeAnalysis;
+using Nox.Generator.Common;
+using Nox.Solution;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Nox.Generator.Application;
+
+internal class DomainEventHandlerGenerator : INoxCodeGenerator
+{
+    public NoxGeneratorKind GeneratorKind => NoxGeneratorKind.Application;
+
+    public void Generate(SourceProductionContext context, NoxSolutionCodeGeneratorState codeGeneratorState, GeneratorConfig config)
+    {
+        context.CancellationToken.ThrowIfCancellationRequested();
+
+        if (codeGeneratorState.Solution.Infrastructure?.Messaging is null || codeGeneratorState.Solution.Domain?.Entities is null)
+            return;
+
+        foreach (var (crudOperation, raiseIntegrationEvent, entity) in GroupEntitiesWithDomainEventsByCrudOperation(codeGeneratorState.Solution.Domain.Entities))
+        {
+            context.CancellationToken.ThrowIfCancellationRequested();
+
+            new TemplateCodeBuilder(context, codeGeneratorState)
+                .WithClassName($"{entity.Name}{crudOperation}DomainEventHandler")
+                .WithFileNamePrefix($"Application.DomainEventHandlers")
+                .WithObject("crudOperation", crudOperation)
+                .WithObject("raiseIntegrationEvent", raiseIntegrationEvent)
```

Comment:
skip if we are not raising this object is unnecessary

Reply:
Changes were applied.